opm/simulators/flow/DamarisWriter.cpp

```diff
@@ -42,7 +42,8 @@ int setPosition(const char* field, int rank, int64_t pos)
 {
     int dam_err = damaris_set_position(field, &pos);
     if (dam_err != DAMARIS_OK) {
-        OpmLog::error(fmt::format("damariswriter::writeOutput()       : ( rank:{})"
+        // OpmLog::error only prints from rank 0
+        OpmLog::error(fmt::format("damariswriter::setPosition()       : ( rank:{}) "
```

Comment:
I think we must use the DeferredLogger class here to ensure we get all error log messages. Then use gatherDeferredLogger() to gather them to rank 0 once we are in a context where every process will be. Tell me if you want some guidance on this.

Also: The `error` category should only be for things that stop the simulator. If this is not necessary (and I do not think it is, no exception is thrown and the process will keep running), use the `problem` (for numerical issues) or `warning` categories instead. Probably `warning` is the best here.

Reply:
The test:
```
const bool data_is_integer = (name == "MPI_RANK" || name == "GLOBAL_CELL_INDEX");
```
 will not allow a future developer/user to add a variable that is of integer type. I assume that because a std::variant is used, that integer data vectors are possible.

Reply:
hanged to warning. removed comments about only rank 0 getting the error/warning message - this still seems an issue as other ranks can have an error and the error message will be hidden.